### PR TITLE
terraform support for manual app registration

### PR DESCRIPTION
**Pull request by asbanger**
This pull request primarily focuses on enabling manual app registration for the Information Assistant web application. The changes involve the addition of a comprehensive guide for manual app registration in the `docs/deployment/manual_app_registration.md` file and the introduction of optional environment variables in the `scripts/inf-create.sh` script to accommodate manual app registration details.

Manual App Registration Guide:

* [`docs/deployment/manual_app_registration.md`](diffhunk://#diff-a00bd32fb9c4cb67f5fe5c8923fcbf76ea5d62b5a11b3adeb423ebd7285f83d7R1-R92): A detailed guide was added to assist in manual app registration if the user doesn't have tenant-level permissions. This guide includes steps to create Azure AD App Registrations, obtain necessary information from the tenant administrator, and adjust the code in the infrastructure deployment.

Script Changes:

* [`scripts/inf-create.sh`](diffhunk://#diff-bd154d4408caa1f5f6980ad6076c55205a73f1328b63e56e759dd56b6a654e1eR63-R70): Optional environment variables were introduced to accommodate details from manual app registration. These variables can be set if the user has created Azure AD App Registrations and doesn't have tenant-level permissions.

### CODE REVIEW

File: docs/deployment/manual_app_registration.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+export TF_VAR_isInAutomation=true
+export TF_VAR_aadWebClientId=""
+export TF_VAR_aadMgmtClientId=""
+export TF_VAR_aadMgmtServicePrincipalId=""
+export TF_VAR_aadMgmtClientSecret=""
+
+- Management Access App Registration Client Secret
+
+### 4.Adjust Code in Infrastructure Deployment
```
Comment (dayland) [inline docs/deployment/manual_app_registration.md:76]: There are two #4 sections that appear to be duplicate.

File: docs/deployment/manual_app_registration.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+- Management Access Service Principal ID (guid)
+- Management Access App Registration Client Secret (string)
+
+These values will be used to update the code in the Infrastructure Deployment section.
+
+### 4.  Adjust Code in Infrastructure Deployment
+
+In the file `scripts/inf-create.sh`, between lines 63 - 69, you would need to uncomment the code  and update parameters with values provided by your tenant Administrator.
+
+export TF_VAR_isInAutomation=true
```
Comment (dayland) [inline docs/deployment/manual_app_registration.md:68]: This should be in code blocks not plain text.

File: docs/deployment/manual_app_registration.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+export TF_VAR_aadWebClientId=""
+export TF_VAR_aadMgmtClientId=""
+export TF_VAR_aadMgmtServicePrincipalId=""
+export TF_VAR_aadMgmtClientSecret=""
+
+- Management Access App Registration Client Secret
```
Comment (dayland) [inline docs/deployment/manual_app_registration.md:74]: What is this extra bullet for, doesn't seem to fit here?

File: docs/deployment/manual_app_registration.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
+# App Registration Creation Guide
```
Comment (dayland) [inline docs/deployment/manual_app_registration.md:1]: How will someone find this documentation? This page is just in a random folder. I recommend linking to it from the Optional Features page.

**Review by dayland — CHANGES_REQUESTED** [review]

File: docs/deployment/manual_app_registration.md (hunk 1 above)
Comment (asbanger) [reply docs/deployment/manual_app_registration.md:76]: Looks like i tried fix using co-pilot, it created duplicate entry. Fixed it

File: docs/deployment/manual_app_registration.md (hunk 3 above)
Comment (asbanger) [reply docs/deployment/manual_app_registration.md:74]: Fixed it

File: docs/deployment/manual_app_registration.md (hunk 2 above)
Comment (asbanger) [reply docs/deployment/manual_app_registration.md:68]: Fixed it

File: docs/deployment/manual_app_registration.md (hunk 4 above)
Comment (asbanger) [reply docs/deployment/manual_app_registration.md:1]: Link back to deployment procedure .

Comment (asbanger) [reply docs/deployment/manual_app_registration.md:1]: Completed

**Review by dayland — APPROVED** [review]

File: docs/deployment/deployment.md [hunk 5]
```diff
@@ -166,6 +166,10 @@ run-data-migration           Run the data migration moving data from one resourc
 manual-inf-destroy           A command triggered by a user to destroy a resource group, associated resources, and related Entra items
 ```
 
+## Configure AD app registration ( manual steps )
+
+If you have insufficient permissions at the tenant level (Application Administrator Entry Role), follow the guide to complete the deployment [manual app registration](/docs/deployment/manual_app_registration.md).
```
[the remark this answers is not in the document]
Comment (asbanger) [reply docs/deployment/deployment.md]: Fixed the typo.

**Review by dayland — APPROVED** [review]

**Pull request merged**